clients/src/main/java/org/apache/kafka/clients/producer/internals/TransactionManager.java

```diff
@@ -606,27 +604,44 @@ Priority priority() {
         @Override
         public void handleResponse(AbstractResponse response) {
             AddPartitionsToTxnResponse addPartitionsToTxnResponse = (AddPartitionsToTxnResponse) response;
-            Errors error = addPartitionsToTxnResponse.error();
-            if (error == Errors.NONE) {
+            Map<TopicPartition, Errors> errors = addPartitionsToTxnResponse.errors();
+            boolean hasPartitionErrors = false;
+            for (TopicPartition topicPartition : pendingPartitionsToBeAddedToTransaction) {
```

Comment:
Hmm. I don't think this is the right structure for this handler. Most of the errors are global, and handling them on a per partition basis could be inefficient.

Specifically, the coordinator errors like `COORDINATOR_NOT_AVAILABLE`, `COORDINATOR_LOAD_IN_PROGRESS`, etc. would be set for all partitions in the request. So we will needlessly enqueue multiple copies of the FindCoordinatorRequest for each partition, and send all these extra requests out. This should not happen.